eox_nelp/pearson_vue/rti_backend.py

```diff
@@ -47,9 +54,20 @@ def run_pipeline(self):
 
         for idx, func in enumerate(pipeline[pipeline_index:]):
             self.backend_data["pipeline_index"] = pipeline_index + idx
-            result = func(**self.backend_data) or {}
-            self.backend_data.update(result)
+            try:
+                result = func(**self.backend_data) or {}
+            except PearsonBaseError as pearson_error:
+                # clean kwargs to dont finish next pipeline launch.
+                executed_pipeline_kwargs = remove_keys_from_dict(self.backend_data, ["pipeline_index"])
+                tasks = importlib.import_module("eox_nelp.pearson_vue.tasks")
+                tasks.rti_error_handler_task.delay(
+                    failed_step_pipeline=func.__name__,
+                    exception_data=pearson_error.__dict__,
+                    **executed_pipeline_kwargs,
+                )
+                break
```

Comment:
I would let you know the intention of who needs the data??. And the dynamic kwargs behaviour.

Let's Imagine a case: I  want to use  `profile_metadata` in the error pipeline.

As the error_task is triggered when any step fails I don't know the kwargs executed(pipeline_step) in the initial RTI pipeline.
For eg 
```  python

    def get_pipeline(self):
        """
        Returns the RTI pipeline, which is a list of functions to be executed.
        """
        return [
            handle_course_completion_status,
            get_user_data,
            get_exam_data,
            build_cdd_request,
            validate_cdd_request,
            build_ead_request,
            validate_ead_request,
            check_service_availability,
            import_candidate_demographics,
            import_exam_authorization,
        ]
```
Here we have pipe with different inputs and outputs.
So I would add all kwargs definition like:
``` python3
tasks.rti_error_handler_task.delay(
                    failed_step_pipeline=func.__name__,
                    exception_data=pearson_error.__dict__,
                   course_id=self.backend_data.get("course_id"),
                    user_id=self.backend_data.get("user_id")
                    profile_metadata=self.backend_data.get("profile_metadata"),
                   exam_metadata=self.backend_data.get("exam_metadata"),
                 ead_request=self.backend_data.get("ead_request"),
                 cdd_request=self.backend_data.get("cdd_request"),
                transaction_type=self.backend_data.get("transaction_type"),
                ...
                )
```
And also here you have to add any  other possible return of the pipeline steps(new steps)...


Or only create a pipeline with 
``` python
     if not  kwargs.get("profile_metadata"):
         return
```

Now, using **kwargs, when launching the error pipeline you only specify that this pipeline is like the continuation of the first pipeline so it would have the same kwargs.(executed-dynamic~)